### ISSUE TITLE
Expose a library to allow integration in TUI

### ISSUE THREAD

**Issue by joseluis** [atanunq/viu#9]
It could be useful to be able to use viu as an image viewer for terminal apps. E.g.: https://github.com/fdehau/tui-rs/issues/143

**Comment by atanunq** [comment]
Thanks for your suggestion. I will definitely consider doing that later on. For now I will focus on polishing it and adding support for more terminals/platforms as right now it is not far from the "works on my machine" joke.

**Comment by adelarsq** [comment]
Just for reference: [kitty terminal](https://sw.kovidgoyal.net/kitty/graphics-protocol.html)

**Comment by jarun** [comment]
Excellent project!

I have just added 2 plugins in [nnn](https://github.com/jarun/nnn) to view an image or a directory of images using viu.

**Comment by joseluis** [comment]
Very nice project @jarun. I like the simplicity of its plugin system. It calls viu as a shell command.

I would like to be able to show images from a buffer in memory also without going through the file system.

**Comment by jarun** [comment]
> I would like to be able to show images from a buffer in memory also without going through the file system.

I believe that's this issue is for, am I correct? As long as it can be done using a script, the `nnn` scripts can be updated to do that.

**Comment by atanunq** [comment]
Thank you for saying this and making the plugins, @jarun! I have been following your work for a while now and you really are an inspiration when it comes to user interfaces in the terminal!

**Comment by jarun** [comment]
Thank you for the compliment! I'm flattered! :dancing_men:

**Comment by mrusme** [comment]
Definitely looking forward to a libviu, if that's still going to happen! Could use this a lot across different projects.

**Comment by atanunq** [comment]
Thank you all for the feedback!
I extracted `viu`'s image printing functionality into a library, called [`viuer`](https://github.com/atanunq/viuer). I just released version 0.1 and would love to hear your thoughts. It supports only block printing for now, but the foundation has been laid.

Just fyi, `viu` now uses it in the [`external-library`](https://github.com/atanunq/viu/tree/external-library) branch and looks good to me. More testing will be required before I merge it in master, though.
